tables/DataMan/DataManager.cc

```diff
@@ -316,7 +317,8 @@ DataManagerCtor DataManager::getCtor (const String& type)
         libname = libname.substr (0, pos);
     }
     // Try to load and initialize the dynamic library.
-    DynLib dl(libname, string("libcasa_"), "register_"+libname, False);
+    DynLib dl(libname, string("libcasa_"), STRING(CASACORE_MAJOR_VERSION),
```

Comment:
Do you propose to use the SOVERSION specified in CMakeLists.txt, thus passing it in as a compile variable? It should be the same as the versions defined in version.h, but I agree it can be forgotten to update version.h. So what is best?
Ole suggested to only use the major version, not minor and patch, but it is no problem to do so.

I'll improve the error message. The error messages from dlopen show the paths and versions tried (which become part of the error message), thus there is no need to include it in the error message.